TicTacToe.cpp

```diff
@@ -0,0 +1,250 @@
+   if (!unentschieden){
+      std::cout << ausgabe << " ist der Gewinner.";
+   }
+   else{
+      std::cout << "Das Spiel ist unentschieden ausgegangen.";
+   }   
+   return 0;
+}
+
+bool gameover(std::vector<std::string> &feld) {
+   if (zuege > 9)
+   {
+      return true;
+   }
+   else
+   {
+      bool var = false;
+      // Überprüfen der einzelnen Zeilen
+      // Überprüfen Zeile 1
+      for (int i = 0; i < 3; i++) //Zeilen überprüfen
+      {
+         var = checkline(i, feld);
+         if (var)
+         {
+            return true;
+         }
+      }
+      for (int j = 0; j < 3; j++) //splitn überprüfen
+      {
+         var = checksplit(j, feld); 
+         if (var)
+         {
+            return true;
+         }
+      }
+      var = checkMainDiagonal(feld); // 1. Diagonale wird überprüft
+      if (var)
+      {
+         return true;
+      }
+      var = checkSecondaryDiagonal(feld); // 2. Diagonale wird überprüft
+      if (var)
+      {
+         return true;
+      }
+   }
+}
+
+//findTheWinner: Funktion, die Gewinner ermittelt und zurückgibt
+
+char findTheWinner(std::vector<std::string> &feld, bool &unentschieden) {
+   if (zuege <= 9)
+   {
+      bool var = false;
+      // Überprüfen der einzelnen Zeilen
+      // Überprüfen Zeile 1
+      for (int i = 0; i < 3; i++) //Zeilen überprüfen
+      {
+         var = checkline(i, feld);
+         if (var)
+         {
+            return feld[i][0];
+         }
+      }
+      for (int j = 0; j < 3; j++) //splitn überprüfen
+      {
+         var = checksplit(j, feld);
+         if (var)
+         {
+            return feld[0][j];
+         }
+      }
+      var = checkMainDiagonal(feld);
+      if (var)
+      {
+         return feld[0][0];
+      }
+      var = checkSecondaryDiagonal(feld);
+      if (var)
+      {
+         return feld[2][2];
+      }
+   }
+   unentschieden = true;
+   return 'O';
+}
+
+//Überprüft ob eine Zeile nur gleiche Elemente enthält
+
+bool checkline (int zeile,std::vector<std::string> &feld )
+{
+   if (feld[zeile][0] == feld[zeile][1] && feld[zeile][1] == feld[zeile][2]) 
+   {
+      return true;
+   }
+   else
+   {
+      return false;
+   }
+}
+
+//Überprüft, ob eine Spalte nur gleiche Elemente enthält
+
+bool checksplit(int split,std::vector<std::string> &feld)
+{
+   if (feld[0][split] == feld[1][split] && feld[1][split] == feld[2][split])
+   {
+      return true;
+   }
+   else
+   {
+      return false;
+   }
+}
+
+//MainDiagonal: von Links Oben nach rechts unten
+
+bool checkMainDiagonal(std::vector<std::string> &feld)
+{
+   if (feld[0][0] == feld[1][1] && feld[1][1] == feld[2][2])
+   {
+      return true;
+   }else
+   {
+      return false;
+   }
+}
+
+//SecondaryDiagonal: von Rechts Oben, nach Links unten
+
+bool checkSecondaryDiagonal(std::vector<std::string> &feld)
+{
+   if (feld[0][2] == feld[1][1] && feld[1][1] == feld[2][0])
+   {
+      return true;
+   }
+   else
+   {
+      return false;
+   }
+}
```

Comment:
Könnte man zusammen fassen durch Verwendung von logischen Verknüpfungen (`||`, `&&`, etc)
```suggestion
bool checkMainDiagonal(std::vector<std::string> &feld)
{
   if ((feld[0][0] == feld[1][1] && feld[1][1] == feld[2][2]) || (feld[0][2] == feld[1][1] && feld[1][1] == feld[2][0]))
   {
      return true;
   }else
   {
      return false;
   }
}
//SecondaryDiagonal: von Rechts Oben, nach Links unten
```

Oder ganz kurz mit tenärem Operator:
```
return ((feld[0][0] == feld[1][1] && feld[1][1] == feld[2][2]) || (feld[0][2] == feld[1][1] && feld[1][1] == feld[2][0])) ? : false;
```
Oder auch nur das Statement, es ist ja schon ein boolscher Ausdruck also wahr oder falsch.